servicetest/agent/test_agent.py

```diff
@@ -93,23 +94,41 @@ def test_old_containers_cleanup(self):
             The second phase of the test is to run SC again and make sure
             it does not fail because of any files left from the previous run.
         """
+        # Create a container and kill SC-agent
         sc = Container()
         container_id = sc.start(DATA)
+        container_name = "SC-" + str(container_id)
         os.system("pkill --signal SIGKILL -f softwarecontainer-agent")
         time.sleep(0.5)
 
-        os.system("lxc-ls >> output")
-        with open("output", "r") as fh:
-            file_content = fh.readline()
-        file_content = file_content.strip()
-        assert file_content == ("SC-" + str(container_id))
-        os.remove("output")
+        skipped_container_name = "agent-servicetest"
+        # Create one that SC shouldn't touch
+        os.system("lxc-create -t busybox -n " + skipped_container_name)
 
+        # Check the content of lxc-ls
+        existing_containers = []
+        proc = subprocess.Popen(["lxc-ls", "-1"], stdout=subprocess.PIPE)
+        for line in proc.stdout:
+            existing_containers.append(line.strip())
+        assert proc.wait() == 0
+
+        assert existing_containers.count(container_name) == 1
+        assert existing_containers.count(skipped_container_name) == 1
+
+        # Start the agent and give it time to kill the created container
         agent_handler = SoftwareContainerAgentHandler(logfile_path(), None, None)
         time.sleep(0.5)
-        os.system("lxc-ls >> output")
-        with open("output", "r") as fh:
-            file_content = fh.readline()
-        assert file_content == ""
+
+        # Check the content of lxc-ls
+        existing_containers = []
+        proc = subprocess.Popen(["lxc-ls", "-1"], stdout=subprocess.PIPE)
+        for line in proc.stdout:
+            existing_containers.append(line.strip())
+        assert proc.wait() == 0
+
+        assert existing_containers.count(container_name) == 0
+        assert existing_containers.count(skipped_container_name) == 1
+
+        # Exit and cleanup
         agent_handler.terminate()
-        os.remove("output")
+        os.system("lxc-destroy -n " + skipped_container_name)
```

Comment:
This cleanup will be skipped it there is an exception (or assertion error) in the code above.